contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,569 @@
+  covered in this proposal.
+  
+  The last design point is block devices should be able to be fully restricted by the admin in accordance with how inline volumes 
+  are today. Ideally, the admin would want to be able to restrict either raw-local devices and or raw-network attached devices.
+  
+  To ensure backwards compatibility and a phased transition of this feature, the consensus from the community is to intentionally disable
+  the volumeMode: Block for external provisioners until a suitable implementation for provisioner versioning has been accepted and 
+  implemented in the community. This requirement is better described in the design PR discussion and will be implemented as a seperate
+  initiative. Acceptable values for volumeMode are 'Block' and 'Filesystem'. Where 'Filesystem' is the default value today and not 
+  required to be set in the PV/PVC.
+  
+# Proposed API Changes
+   
+## Persistent Volume Claim API Changes:
+In the simplest case of static provisioning, a user asks for a volumeMode of block. The binder will only bind to a PV defined 
+with the same volumeMode.
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+name: myclaim
+spec:
+  volumeMode: Block #proposed API change
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 80Gi 
+```
+
+For dynamic provisioning and the use of the storageClass, the admin also specifically defines the intent of the volume by 
+indicating the volumeMode as block. The provisioner for this class will validate whether or not it supports block and return
+an error if it does not.
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+name: myclaim
+spec:
+  storageClassName: local-fast 
+  volumeMode: Block #proposed API change
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 80Gi 
+```
+
+## Persistent Volume API Changes:
+For static provisioning the admin creates the volume and also is intentional about how the volume should be consumed. For backwards
+compatibility, the absence of volumeMode will default to file which is how volumes work today, which are formatted with a filesystem depending on the plug-in chosen. Recycling will not be a supported reclaim policy as it has been deprecated. The path value in the local PV definition would be overloaded to define the path of the raw block device rather than the fileystem path.
+```
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+name: local-raw-pv
+spec:
+  volumeMode: Block #proposed API change
+  capacity:
+    storage: 100Gi
+  local:
+    path: /dev/xvdc #device path
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Delete 
+```
+## Pod API Changes:
+To provide better specificy and ensure support of inline volumes, the following changes are proposed in the pod specification.
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-db
+spec:
+    containers
+    - name: mysql
+      image: mysql
+      volumeDevices:
```

Comment:
Add a comment here showing that this is a new API change

Reply:
done